src/task.js

```diff
@@ -2,30 +2,26 @@ import { allProjects } from "./project";
 
 // constructor for new tasks
 function task(title, description) {
-    this.title = title,
-    this.description = description
+  (this.title = title),
+    (this.description = description),
+    (this.checked = false);
 }
 
 export const taskModal = () => {
-    let modal = document.querySelector('#taskModal')
-    modal.showModal()
-}
+  let modal = document.querySelector("#taskModal");
+  modal.showModal();
+};
 
-export const submitTask = () => {
-    // Grabs the value of these inputs and creates task with my constructor
-    let taskName = document.getElementById('taskName')
-    let taskDescription = document.getElementById('taskDescription')
-    let taskInfo = new task(taskName.value, taskDescription.value)
-    console.log(taskInfo)
+export const submitTask = (project) => {
+  // Grabs the value of these inputs and creates task with my constructor
+  let taskName = document.getElementById("taskName");
+  let taskDescription = document.getElementById("taskDescription");
+  let taskInfo = new task(taskName.value, taskDescription.value);
+  console.log(taskInfo);
 
-    let currentProject = undefined
-    // Finds my active project and appends only into that object
-    for (let i = 0; i < allProjects.length; i++){
-        currentProject = allProjects.find(object => object.element.className == 'active')
-    }
-    currentProject.tasks.push(taskInfo)
```

Comment:
This is over complicated. It is not convenient to look for the current working element everytime you do something.

This is good that you understand that you can do this. It is useful in a lot of situation. But here there is a problem with the structure of your code. Instead of looking for the current project here, the function should ask for the current project. The caller would have to provide this (it lies in the state).